### PR TITLE
Add the juniper mx

### PR DESCRIPTION
**Pull request by lindycoder**

### CODE REVIEW

**Comment by lindycoder** [comment]
STILL IN THE WORKINGS

**Comment by lindycoder** [comment]
Ready for reviewing

File: fake_switches/netconf/__init__.py [hunk 1]
```diff
@@ -109,11 +110,32 @@ def __init__(self, msg, severity="error", err_type=None, tag=None, info=None, pa
         self.info = info
         self.path = path
 
+    def to_dict(self):
+        return error_to_rpcerror_dict(self)
+
+    def to_etree(self):
+        return dict_2_etree(self.to_dict())
+
```
Comment (fbouliane) [inline fake_switches/netconf/__init__.py]: let it breathe

Comment (lindycoder) [reply fake_switches/netconf/__init__.py]: breath has been given

https://cdn.movieweb.com/img.news.tops/NEfArFf7LqO8il_3_b/Frozen-Movie-Lawsuit-Disney-Idina-Menzel-Demi-Lovato.jpg

File: tests/juniper_mx/juniper_mx_protocol_test.py [hunk 2]
```diff
@@ -0,0 +1,1123 @@
...
+
+        ge001 = self.get_interface("xe-0/0/1")
+        assert_that(ge001, is_(None))
+
+        ge002 = self.get_interface("xe-0/0/2", )
+        assert_that(ge002, is_(None))
+
+        self.cleanup(vlan("VLAN2995"), reset_interface("ae1"), reset_interface("xe-0/0/1"), reset_interface("xe-0/0/2"))
+
+    def test_auto_negotiation_and_no_aut0_negotiation_are_mutually_exclusive(self):
```
Comment (fbouliane) [inline tests/juniper_mx/juniper_mx_protocol_test.py]: aut0 comment dans zer0

Comment (lindycoder) [reply tests/juniper_mx/juniper_mx_protocol_test.py]: will fix in upcoming review

**Pull request merged**